app/scripts/components/exploration/hooks/use-analysis-data-request.ts

```diff
@@ -0,0 +1,98 @@
+import { useCallback, useEffect } from 'react';
+import { useQueryClient } from '@tanstack/react-query';
+import { FeatureCollection, Polygon } from 'geojson';
+import { PrimitiveAtom, useAtom, useAtomValue } from 'jotai';
+
+import { requestDatasetTimeseriesData } from '../analysis-data';
+import { analysisControllerAtom, selectedIntervalAtom } from '../atoms/atoms';
+import { useTimelineDatasetAnalysis } from '../atoms/hooks';
+import { analysisConcurrencyManager } from '../concurrency';
+import { TimelineDataset, TimelineDatasetStatus } from '../types.d.ts';
+import useAois from '$components/common/map/controls/hooks/use-aois';
+
+export function useAnalysisController() {
+  const [controller, setController] = useAtom(analysisControllerAtom);
+
+  return {
+    analysisRun: controller.runId,
+    isAnalyzing: controller.isAnalyzing,
+    isObsolete: controller.isObsolete,
+    setObsolete: useCallback(
+      () => setController((v) => ({ ...v, isObsolete: true })),
+      [] // eslint-disable-line react-hooks/exhaustive-deps -- setController is stable
+    ),
+    runAnalysis: useCallback(
+      () =>
+        setController((v) => ({
+          ...v,
+          runId: v.runId + 1,
+          isAnalyzing: true,
+          isObsolete: false
+        })),
+      [] // eslint-disable-line react-hooks/exhaustive-deps -- setController is stable
+    ),
+    cancelAnalysis: useCallback(
+      () =>
+        setController((v) => ({
+          ...v,
+          runId: 0,
+          isAnalyzing: false,
+          isObsolete: false
+        })),
+      [] // eslint-disable-line react-hooks/exhaustive-deps -- setController is stable
+    )
+  };
+}
+
+export function useAnalysisDataRequest({
+  datasetAtom
+}: {
+  datasetAtom: PrimitiveAtom<TimelineDataset>;
+}) {
+  const queryClient = useQueryClient();
+
+  const selectedInterval = useAtomValue(selectedIntervalAtom);
+  const { features } = useAois();
+  const selectedFeatures = features.filter((f) => f.selected);
+
+  const { analysisRun } = useAnalysisController();
+
+  const dataset = useAtomValue(datasetAtom);
+  const datasetStatus = dataset.status;
+
+  const [, setAnalysis] = useTimelineDatasetAnalysis(datasetAtom);
```

Comment:
You could have `useTimelineDatasetAnalysis` only return a setter (`useSetAtom`) and simplify this:
```suggestion
  const setAnalysis = useTimelineDatasetAnalysis(datasetAtom);
```

Reply:
that's a good point. I was trying to be consistent with the other `useTimelineDataset*`, like `useTimelineDatasetVisibility` or `useTimelineDatasetSettings` where they return getters and setters. You think it is not needed?

Reply:
The consistency argument is a reasonable one, but in that case why not just returning the atoms themselves?

```
export function useTimelineDatasetAnalysisAtom() {
  const analysisAtom = ...

  return analysisAtom;
}
```

```suggestion
  const setAnalysis = useSetAtom(useTimelineDatasetAnalysisAtom(datasetAtom));
```

Reply:
I was trying to avoid this function train and make the hook usage more direct. Nevertheless I'm not set on this format and I'm not opposed to changing it. Do you have a preference?